### DISCUSSION TITLE
Debugging Extensions in Remote (Tunnels/SSH)

### DISCUSSION THREAD

**Discussion by cheetah-darst** [microsoft/vscode-discussions#2282] [Extension Development QnA] [answered]
Hello

I am working on a plugin that will be used primarily when connected to a Remote Host. My team uses the **Remote - SSH** plugin to connect to our development server. I am running into some issues that I am only able to reproduce on the remote connection and I am unable to get the Debugging process to run correctly when trying to connect remotely.

My process is:
* Open extension code in vscode and begin debugging (this opens new vscode window)
* From there I use the Remote Explorer to connect to my remote code
* Back in the original vscode window the debug console reads: "explorer disposed"

And then my debugging session is just sitting there and no longer provides
Has anyone else experienced this?

I imagine I just need to update my launch.json file to have the vscode instance that the debugger opens be a remote session, but I a not sure how to accomplish that.

Thanks

**Comment by olguzzar** [comment, marked as answer]
Hi @cheetah-darst, have you seen the docs about [debugging extensions in a remote environment](https://code.visualstudio.com/api/advanced-topics/remote-extensions#debugging-extensions)?

**Reply by burkeholland** [reply]
Marking this as the answer and closing for now due to lack of activity. @cheetah-darst if you think this is a mistake, please feel free to re-open. Thank you!
